src/InteractiveWindow/EditorTest/InteractiveWindowTests.cs

```diff
@@ -1109,6 +1120,79 @@ public void SelectAllInHeader()
+        {
+            var original = new InteractiveWindowTextSnapshot(_testHost);
+            Window.AddToHistory("");
+
+            var actual = new InteractiveWindowTextSnapshot(_testHost);
+            InteractiveWindowTextSnapshot.AssertEqual(original, actual);
+        }
+
+        [WpfFact]
+        public void AddToHistory_NewLine() 
+        {
+            var original = new InteractiveWindowTextSnapshot(_testHost);
+            Window.AddToHistory(Environment.NewLine);
+
+            var actual = new InteractiveWindowTextSnapshot(_testHost);
+            var expected = original
+                .InsertBeforeLastPrompt(_testHost.Evaluator.GetPrompt(), ReplSpanKind.Prompt)
+                .InsertBeforeLastPrompt(Environment.NewLine, ReplSpanKind.Input);
+
+            InteractiveWindowTextSnapshot.AssertEqual(expected, actual);
+        }
+
+        [WpfFact]
+        public async Task AddToHistory_BetweenWrites() 
+        {
+            var original = new InteractiveWindowTextSnapshot(_testHost);
+
+            await Task.Run(() => Window.WriteLine("a"));
+            await Task.Run(() => Window.WriteLine("b"));
+
+            Window.AddToHistory("c");
+
+            await Task.Run(() => Window.WriteLine("d"));
+            await Task.Run(() => Window.WriteLine("e"));
+
+            Window.FlushOutput();
+
+            var actual = new InteractiveWindowTextSnapshot(_testHost);
+            var expected = original
+                .InsertBeforeLastPrompt($"a{Environment.NewLine}b{Environment.NewLine}", ReplSpanKind.Output)
+                .InsertBeforeLastPrompt(_testHost.Evaluator.GetPrompt(), ReplSpanKind.Prompt)
+                .InsertBeforeLastPrompt($"c{Environment.NewLine}", ReplSpanKind.Input)
+                .InsertBeforeLastPrompt($"d{Environment.NewLine}e{Environment.NewLine}", ReplSpanKind.Output);
+
+            InteractiveWindowTextSnapshot.AssertEqual(expected, actual);
+        }
+
+        [WpfFact]
+        public void AddToHistory_HasInput()
+        {
+            var original = new InteractiveWindowTextSnapshot(_testHost);
+
+            Window.Operations.TypeChar('a');
+            Window.Operations.TypeChar('b');
+            Window.AddToHistory("c");
+            Window.Operations.TypeChar('d');
+            Window.Operations.TypeChar('e');
+            Window.AddToHistory("f");
+
+            var actual = new InteractiveWindowTextSnapshot(_testHost);
+            var expected = original
+                .InsertBeforeLastPrompt(_testHost.Evaluator.GetPrompt(), ReplSpanKind.Prompt)
+                .InsertBeforeLastPrompt($"c{Environment.NewLine}", ReplSpanKind.Input)
+                .InsertBeforeLastPrompt(_testHost.Evaluator.GetPrompt(), ReplSpanKind.Prompt)
+                .InsertBeforeLastPrompt($"f{Environment.NewLine}", ReplSpanKind.Input)
+                .AddToTheEnd($"abde", ReplSpanKind.Input);
+
+            InteractiveWindowTextSnapshot.AssertEqual(expected, actual);
+        }
```

Comment:
Could you please also add a couple of tests to check if the command inserted through `AddToHistory` is also added to navigable command history correctly?